app/classes/submissions_importer.rb

```diff
@@ -37,120 +37,135 @@ def self.attribute_val(row, attributes, name)
     row['f'][attributes.index(name)]['v']
   end
 
-  def self.create_submissions(data, test_type)
-    puts "Importing #{data.count} #{test_type}s"
-
-    data.each do |row|
-      count = Submission.unscoped.where('test_date = ? AND ip_address = ? AND test_type = ?', Date.parse(row[:UTC_date_time]), row[:client_ip], test_type).count
-      next if count > 0
-
-      submission = Submission.new
-      submission.from_mlab           = true
-      submission.completed           = true
-      submission.test_type           = test_type
-      submission.ip_address          = row[:client_ip]
-      submission.test_date           = row[:UTC_date_time]
-      submission.address             = row[:city]
-      submission.area_code           = row[:area_code]
-      submission.zip_code            = row[:postal_code]
-      submission.hostname            = row[:client_hostname]
-      submission.latitude            = row[:client_latitude]
-      submission.longitude           = row[:client_longitude]
-      submission.provider            = Submission.provider_mapping(submission.get_provider)
-      submission.actual_down_speed   = row[:downloadThroughput]
-      submission.actual_upload_speed = row[:uploadThroughput]
-      submission.census_status       = Submission::CENSUS_STATUS[:pending]
-
-      submission.save
-    end
-  end
-
   def self.import
     client = bigquery_init
 
-    zips = ZipBoundary.pluck(:name)
-    zip_codes = "'#{zips.join("','")}'"
+    country_code = 'US'
+    regions = ['OR', 'WA', 'ID']
+    end_time = Date.today.strftime("%Y-%m-%d")
+
+    regions.each do |region|     
 
-    upload_query = upload_query(zip_codes)
-    download_query = download_query(zip_codes)
+      while true #we may need to get multiple batches
```

Comment:
Nothing to see here... 🚶 💨

Reply:
Does Rails do asynchronous calls, or does this go quickly such that it doesn't block the main thread?

Reply:
The Submission Import runs in its own container/runtime. So, it doesn't really block anything as that container is only handling data import/processing.